### PR TITLE
GH844: Added working directory support for tools

### LINKED ISSUE #844: Add working directory to tool settings

### PR DESCRIPTION
**Pull request by patriksvensson**
Closes #844

### CODE REVIEW

**Comment by patriksvensson** [comment]
I would need @gep13 to review this since the change affects GitReleaseNotes. @devlead is of course welcome with opinions as well :smile:

File: src/Cake.Core/Tooling/Tool.cs [hunk 1]
```diff
@@ -226,6 +226,10 @@ protected IProcess RunProcess(TSettings settings, ProcessArgumentBuilder argumen
         /// <returns>The working directory for the tool.</returns>
         protected virtual DirectoryPath GetWorkingDirectory(TSettings settings)
         {
+            if (settings.WorkingDirectory != null)
+            {
+                return settings.WorkingDirectory;
+            }
             return _environment.WorkingDirectory;
```
Comment (gep13) [inline src/Cake.Core/Tooling/Tool.cs]: @patriksvensson is it just me that would like to see a new line after the closing bracket here?  If so, I will get over it :smile:

**Comment by gep13** [comment]
@patriksvensson one small comment, but aside from that, this looks good to me.  If you want to rebase, I will get this merged in.

**Comment by patriksvensson** [comment]
@gep13 Sure, will fix and rebase.

**Comment by patriksvensson** [comment]
@gep13 Fixed and rebased.

**Pull request merged**